src/golang/lib/repos/sqlite/operator.go

```diff
@@ -282,6 +282,46 @@ func (*operatorReader) GetRelationBatch(
 	return relations, err
 }
 
+func (*operatorReader) GetByEngineIntegrationID(
+	ctx context.Context,
+	integrationID uuid.UUID,
+	DB database.Database,
+) ([]models.Operator, error) {
+	query := fmt.Sprintf(`
+		SELECT DISTINCT %s FROM
+		operator, workflow_dag, workflow_dag_edge
+		WHERE
+		workflow_dag_edge.workflow_dag_id = workflow_dag.id
+		AND (
+			workflow_dag_edge.from_id = operator.id
```

Comment:
I wonder if `from_id` itself is sufficient to capture all operators. Do we ever have an operator that doesn't generate any artifact? I think even an operator that returns nothing will generate an artifact.

Reply:
I didn't closely verify this, but I need to do this to pass the unit tests where we generate a operator with upstream but no downsteam

Reply:
I see. Yeah I think this is fine for now to be safe.